internal/receiver/discoveryreceiver/config.go

```diff
@@ -0,0 +1,162 @@
+}
+
+// Match defines the rules for the desired match type and resulting log record
+// content emitted by the Discovery receiver
+type Match struct {
+	Record    *LogRecord `mapstructure:"log_record"`
+	Strict    string     `mapstructure:"strict"`
+	Regexp    string     `mapstructure:"regexp"`
+	Expr      string     `mapstructure:"expr"`
+	FirstOnly bool       `mapstructure:"first_only"`
+}
+
+// LogRecord is a definition of the desired plog.LogRecord content to emit for a match.
+type LogRecord struct {
+	Attributes map[string]string `mapstructure:"attributes"`
+	Severity   string            `mapstructure:"severity"`
+	Body       string            `mapstructure:"body"`
+}
+
+func (cfg *Config) Validate() error {
+	var err error
+	for rName, rEntry := range cfg.Receivers {
+		if e := rEntry.validate(); e != nil {
+			err = multierr.Combine(err, fmt.Errorf("receiver %q validation failure: %w", rName, e))
+		}
+	}
+
+	if cfg.WatchObservers == nil || len(cfg.WatchObservers) == 0 {
+		err = multierr.Combine(err, fmt.Errorf("`watch_observers` must be defined and include at least one configured observer extension"))
+	}
+
+	return err
+}
+
+func (re *ReceiverEntry) validate() error {
+	if err := re.Status.validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *Status) validate() error {
+	if s == nil {
+		return fmt.Errorf("`status` must be defined and contain at least one `metrics` or `statements` mapping")
+	}
+
+	if len(s.Metrics) == 0 && len(s.Statements) == 0 {
+		return fmt.Errorf("`status` must contain at least one `metrics` or `statements` mapping with at least one of %v", allowedStatusTypes)
+	}
+
+	var err error
+	statusSources := []struct {
+		matches    map[string][]Match
+		sourceType string
+	}{{s.Metrics, "metrics"}, {s.Statements, "statements"}}
+	for _, statusSource := range statusSources {
+		for statusType, statements := range statusSource.matches {
+			if _, ok := allowedStatuses[statusType]; !ok {
+				err = multierr.Combine(err, fmt.Errorf("unsupported status %q. must be one of %v", statusType, allowedStatusTypes))
+				continue
+			}
+			for _, logMatch := range statements {
+				var matchTypes []string
+				if logMatch.Strict != "" {
+					matchTypes = append(matchTypes, "strict")
+				}
+				if logMatch.Regexp != "" {
+					matchTypes = append(matchTypes, "regexp")
+				}
+				if logMatch.Expr != "" {
+					matchTypes = append(matchTypes, "expr")
+				}
+				if len(matchTypes) != 1 {
+					err = multierr.Combine(err, fmt.Errorf(
+						"`%s` status source type `%s` match type validation failed. Must provide one of %v but received %v", statusSource.sourceType, statusType, allowedMatchTypes, matchTypes,
+					))
+				}
+				if e := logMatch.Record.validate(); e != nil {
+					err = multierr.Combine(err, fmt.Errorf(" %q log record validation failure: %w", statusType, e))
+				}
+			}
+		}
+	}
+	return err
+}
+
+func (lr *LogRecord) validate() error {
+	// TODO: supported severity text validation
```

Comment:
I don't mind if you want to use your own impl of LogRecord, but it might be easier to use plog.LogRecord if you want to access all the fields. For example, your LogRecord struct doesn't have SeverityText or SeverityNum. You would get that for free if you use plog.LogRecord.

Reply:
The config log record is used to express the values that are desired in the actual `plog.LogRecord` item that's going to be produced by the receiver. I don't know if it's possible to use a `plog.LogRecord` in its place given the implementation is in "go.opentelemetry.io/collector/pdata/internal/data/protogen/logs/v1" and doesn't use mapstructure, but considering we only care about specific fields with more convenient accessors I don't think it's a concern.

`Severity` becomes `SeverityText` but considering we may want to specify num as well I can update it.